plugins/out_stackdriver/stackdriver.c

```diff
@@ -941,6 +944,98 @@ static msgpack_object *get_payload_labels(struct flb_stackdriver *ctx, msgpack_o
     return NULL;
 }
 
+/*
+ *    pack_resource_labels():
+ *  - Looks through the resource_labels parameter and appends new key value
+ *    pair to the log entry.
+ *  - Supports field access, plaintext assignment and environment variables.
+ */
+static int pack_resource_labels(struct flb_stackdriver *ctx,
+                                        struct flb_mp_map_header *mh,
+                                        msgpack_packer *mp_pck,
+                                        const void *data,
+                                        size_t bytes)
+{
+    struct mk_list *head;
+    struct flb_kv *label_kv;
+    struct flb_record_accessor *ra;
+    struct flb_ra_value *rval;
+    msgpack_object root;
+    msgpack_unpacked result;
+    size_t off = 0;
+    int len;
+
+    if (ctx->should_skip_resource_labels_api == FLB_TRUE) {
+        return -1;
+    }
+
+    len = mk_list_size(&ctx->resource_labels_kvs);
+    if (len == 0) {
+        return -1;
+    }
+
+    msgpack_unpacked_init(&result);
+    if (msgpack_unpack_next(&result, data, bytes, &off) == MSGPACK_UNPACK_SUCCESS) {
+        root = result.data;
+
+        if (!validate_msgpack_unpacked_data(root)) {
+            msgpack_unpacked_destroy(&result);
+            flb_plg_error(ctx->ins, "unexpected record format");
+            return -1;
+        }
+
+        flb_mp_map_header_init(mh, mp_pck);
+        mk_list_foreach(head, &ctx->resource_labels_kvs) {
+            label_kv = mk_list_entry(head, struct flb_kv, _head);
+            /* KVs have the form destination=original, so the original key is the value. */
+            if (label_kv->val[0] == '$') {
```

Comment:
Comment the logic of using dollar sign `$` as the indicator of using the record_accessor or just the plaintext.

Reply:
Done.